core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java

```diff
@@ -0,0 +1,59 @@
+package org.wildfly.extras.sunstone.tests.ec2;
+
+import com.google.common.base.Strings;
+import org.jclouds.aws.ec2.AWSEC2Api;
+import org.jclouds.aws.ec2.domain.AWSRunningInstance;
+import org.jclouds.ec2.domain.Reservation;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.extras.sunstone.api.CloudProperties;
+import org.wildfly.extras.sunstone.api.CloudProvider;
+import org.wildfly.extras.sunstone.api.impl.Config;
+import org.wildfly.extras.sunstone.api.impl.ec2.EC2CloudProvider;
+import org.wildfly.extras.sunstone.api.impl.ec2.EC2Node;
+
+import java.util.Set;
+
+public class EC2SubnetTest {
+
+    @BeforeClass
+    public static void setUpClass() {
+        Assume.assumeFalse(Strings.isNullOrEmpty(System.getProperty("ec2.securityGroupIds")));
+        Assume.assumeFalse(Strings.isNullOrEmpty(System.getProperty("ec2.subnetId")));
+    }
+
+    /**
+     * Starts a node, then queries the AWS EC2 API to find out if the node really belongs to the
+     * subnet that was specified in the .properties file.
+     */
+    @Test
+    public void subnetTest() {
+        CloudProperties.getInstance().reset().load(EC2SubnetTest.class);
+        try (EC2CloudProvider ec2CloudProvider = (EC2CloudProvider) CloudProvider.create("provider0")) {
+            EC2Node ec2Node = (EC2Node) ec2CloudProvider.createNode("node0");
```

Comment:
Ah, true, I forgot that closing the cloud provider closes all non-closed nodes.

I wonder why we always write `try (Node node = cloudProvider.createNode(...)) { ... }` when it really isn't necessary.